src/backend/distributed/commands/alter_table.c

```diff
@@ -1726,6 +1726,17 @@ ReplaceTable(Oid sourceId, Oid targetId, List *justBeforeDropCommands,
 		ExecuteQueryViaSPI(justBeforeDropCommand, SPI_OK_UTILITY);
 	}
 
+	if (!suppressNoticeMessages)
+	{
+		ereport(NOTICE, (errmsg("analyzing the new %s",
+								quote_qualified_identifier(schemaName, sourceName))));
+	}
+
+	resetStringInfo(query);
+	appendStringInfo(query, "ANALYZE %s",
```

Comment:
While I agree that we the table should be analyzed. One thing I'm worried about is that `undistribute_table` is a blocking operation currently, and thus ANALYZE-ing will keep the table locked for a longer period. I feel like its probably better to allow possibly badly optimized queries on an non-analyzed table than to block all writes to it. Especially since usually autovacuum should be analyzing this new table in the background anyway already.

Reply:
Good point and makes sense. Closing the PR then